polaris/ocean/tasks/isomip_plus/topo/remap.py

```diff
@@ -0,0 +1,140 @@
+        Run this step of the test case
+        """
+        self._preprocess()
+        self._remap()
+        self._rename()
+        self._renormalize()
+
+    def _preprocess(self):
+        ice_density = self.config.getfloat('isomip_plus', 'ice_density')
+
+        with (xr.open_dataset('topography.nc') as ds_in):
+            if 't' in ds_in.dims:
+                ds_in = ds_in.chunk({'t': 1})
+            ds_in['iceThickness'] = ds_in.upperSurface - ds_in.lowerSurface
+            ds_in.iceThickness.attrs['description'] = 'ice thickness'
+            ds_in.iceThickness.attrs['units'] = 'm'
+
+            gravity = constants['SHR_CONST_G']
+            ds_in['landIcePressure'] = (ice_density * gravity *
+                                        ds_in.iceThickness)
+            ds_in.iceThickness.attrs['description'] = \
+                'pressure at the ice base'
+            ds_in.iceThickness.attrs['units'] = 'Pa'
+
+            ds_in = ds_in.rename({'floatingMask': 'landIceFloatingFraction',
+                                  'groundedMask': 'landIceGroundedFraction',
+                                  'openOceanMask': 'openOceanFraction',
+                                  'lowerSurface': 'landIceDraft'})
+            ds_in['oceanFraction'] = \
+                (ds_in.bedrockTopography < 0.).astype(float)
+            ds_in['landIceFraction'] = (ds_in.landIceFloatingFraction +
+                                        ds_in.landIceGroundedFraction)
+
+            ds_in.to_netcdf('topography_processed.nc')
+
+    def _remap(self):
+        logger = self.logger
+        topo_map = self.dependencies['topo_map']
+
+        remapper = topo_map.get_remapper()
+
+        remapper.remap_file(inFileName='topography_processed.nc',
+                            outFileName='topography_ncremap.nc',
+                            logger=logger)
+
+    @staticmethod
+    def _rename():
+        with xr.open_dataset('topography_ncremap.nc') as ds_out:
+            drop = ['x', 'y', 'area', 'lat_vertices', 'lon_vertices']
+            rename = {'ncol': 'nCells'}
+            if 't' in ds_out.dims:
+                drop.append('t')
+                rename['t'] = 'Time'
+            ds_out = ds_out.drop_vars(drop)
+            ds_out = ds_out.rename(rename)
+
+            if 'Time' in ds_out.dims:
+                xtime = []
+                for time_index in range(ds_out.sizes['Time']):
+                    time_str = f'{time_index + 1:04d}-01-01_00:00:00'
+                    xtime.append(time_str)
+                ds_out['xtime'] = ('Time', np.array(xtime, 'S64'))
+
+            write_netcdf(ds_out, 'topography_rename.nc')
+
+    @staticmethod
+    def _renormalize():
+        with xr.open_dataset('topography_rename.nc') as ds_out:
+            if 'Time' in ds_out.dims:
+                ds_out = ds_out.chunk({'Time': 1})
+            # renormalize all variables by ocean_frac
+            ocean_frac = ds_out.oceanFraction
+            for var in ds_out:
```

Comment:
Is it really the case that we want to renormalize all the variables in `topography_rename.nc`? I haven't thought about this in detail, but maybe you have already? I have these variables:

```
    variables(dimensions): float64 bedrockTopography(nCells), float64 iceThickness(nCells), float64 landIceDraft(nCells), float64 landIceFloatingFraction(nCells), float64 landIceFraction(nCells), float64 landIceGroundedFraction(nCells), float64 landIcePressure(nCells), float64 oceanFraction(nCells), float64 openOceanFraction(nCells), float64 upperSurface(nCells)
```

In looking at this, I wonder whether we should have dropped `upperSurface`?

Reply:
Yes, I think all variables have to be renormalized to be sensible, even upper surface.  Otherwise, they drop off toward zero in places where ocean fraction is less than 1.  I don't think we currently use upper surface so I'm fine with dropping it.

Reply:
I removed `upperSurface`.